### PR TITLE
IGNITE-14240 Handle authentication error. Fix infinite recursion on failed connection on handshake.

### PR DESCRIPTION
**Pull request by ivandasch**

### CODE REVIEW

File: tests/common/conftest.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import pytest
+
+from pyignite import Client
+from pyignite.api import cache_create, cache_destroy
+from tests.util import start_ignite_gen
+
+
+@pytest.fixture(scope='module', autouse=True)
+def server():
+    yield from start_ignite_gen()
```
Comment (isapego) [inline tests/common/conftest.py:25]: Let's start at least 2 nodes.

File: tests/security/test_auth.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+from tests.util import start_ignite_gen, clear_ignite_work_dir
+
+
+DEFAULT_IGNITE_USERNAME = 'ignite'
+DEFAULT_IGNITE_PASSWORD = 'ignite'
+
+
+@pytest.fixture(scope='module', autouse=True)
+def server(cleanup):
+    yield from start_ignite_gen(use_ssl=True, use_auth=True)
```
[the remark this answers is not in the document]
Comment (ivandasch) [reply tests/security/test_auth.py:28]: Sure, I just thought that creds without ssl is nonsense

File: tests/common/conftest.py (hunk 1 above)
Comment (ivandasch) [reply tests/common/conftest.py:25]: Ok, no problem

File: tests/security/test_ssl.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+def __test_connect_ssl(**kwargs):
+    if 'use_ssl' not in kwargs:
+        kwargs['use_ssl'] = True
+
+    client = Client(**kwargs)
+
+    client.connect("127.0.0.1", 10801)
+
+    assert all(node.alive for node in client._nodes)
```
[the remark this answers is not in the document]
Comment (ivandasch) [reply tests/security/test_ssl.py:45]: Khm, ok, but if handshake succeeded -- everything is ok.

**Pull request closed without merging**